### PR TITLE
"add registry interface"

### LINKED ISSUE #6448: need deperated warning in Python API
During moving from V1 to V2, we need the deprecated warning in Python API.

### PR DESCRIPTION
**Pull request by dzhwinter**
fix #6448

### CODE REVIEW

File: python/paddle/v2/fluid/layers.py [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
+layer_registry('mul')
+layer_registry('elementwise_add')
+layer_registry('elementwise_div')
+layer_registry('dropout')
+layer_registry('reshape')
+layer_registry('sigmoid')
+layer_registry('scale')
+layer_registry('reshape')
+layer_registry('transpose')
+layer_registry('sigmoid_cross_entropy_with_logits')
```
Comment (typhoonzero) [inline python/paddle/v2/fluid/layers.py]: The name should be `register_layer`, it's more meaningful.

File: python/paddle/v2/fluid/layers.py [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
-import cStringIO
+from registry import layer_registry
 from param_attr import ParamAttr
-import contextlib
 
 __all__ = [
     'fc', 'data', 'cross_entropy', 'conv2d', 'pool2d', 'embedding', 'concat',
     'StaticRNN', 'cast', 'sequence_conv', 'sequence_pool', 'sums', 'cos_sim',
     'batch_norm', 'accuracy', 'split_lod_tensor', 'While'
 ]
```
Comment (reyoung) [inline python/paddle/v2/fluid/layers.py]: Maybe we can create an array here.

File: python/paddle/v2/fluid/layers.py (hunk 1 above)
Comment (dzhwinter) [reply python/paddle/v2/fluid/layers.py]: fixed

File: python/paddle/v2/fluid/layers.py (hunk 2 above)
Comment (dzhwinter) [reply python/paddle/v2/fluid/layers.py]: Done.

**Pull request merged**